webapp/src/Service/ExternalContestSourceService.php

```diff
@@ -1468,26 +1468,31 @@ protected function importSubmission(Event $event, EventData $data): void
                     }
 
                     if ($submissionDownloadSucceeded) {
-                        try {
-                            $response = $this->httpClient->request('GET', $zipUrl);
-                            $ziphandler = fopen($zipFile, 'w');
-                            if ($response->getStatusCode() !== 200) {
-                                // TODO: Retry a couple of times.
+                        $tries = 1;
+                        do {
+                            try {
+                                $response = $this->httpClient->request('GET', $zipUrl);
+                                $ziphandler = fopen($zipFile, 'w');
+                                if ($response->getStatusCode() !== 200) {
+                                    $this->addOrUpdateWarning($event, $data->id, ExternalSourceWarning::TYPE_SUBMISSION_ERROR, [
+                                        'message' => "Cannot download ZIP from $zipUrl after trying $tries times",
```

Comment:
Is this the same error we saw with the CDS a couple years back, where they expose a submission before they expose the file?

Reply:
Basically yes. PC2 fixed it on their side now but I think hardening doesn't hurt

Reply:
And that the CCS'es use said extensions 🫠